mindmeld/auto_annotator.py

```diff
@@ -835,5 +839,300 @@ def valid_entity_check(self, entity):
+        Returns:
+            translated_text (str): Translated text
+        """
+        raise NotImplementedError("Subclasses must implement this method")
+
+
+class GoogleTranslator(Translator):
+    """Class for translation using the Google Translate API."""
+
+    def __init__(self):
+        """Initializes the translate_client."""
+        self.translate_client = self.get_translate_client()
+
+    def get_translate_client(self):
+        """Creates a translation client after finding the credential path."""
+        GoogleTranslator._check_credential_exists()
+        return translate_v2.Client()
+
+    @staticmethod
+    def _check_credential_exists():
+        """Searches environment variables for the path to google application credentials.
+
+        Returns:
+            credential_path (str): Path to google application credentials.
+        """
+        try:
+            return os.environ["GOOGLE_APPLICATION_CREDENTIALS"]
+        except KeyError:
+            raise KeyError(
+                "Google credential path not found. Export 'GOOGLE_CREDENTIAL_PATH' as"
+                " an environment variable."
+            )
+
+    def detect_language(self, text):
+        """
+        Args:
+            text (str): Input text
+        Returns:
+            language_code (str): Detected Language Code
+        """
+        return self.translate_client.detect_language(text)["language"]
+
+    def translate(self, text, target_language):
+        """
+        Args:
+            text (str): Input text
+            target_language (str): Language code for language to translate the given text to.
+        Returns:
+            translated_text (str): Translated text
+        """
+        return self.translate_client.translate(text, target_language=target_language)[
+            "translatedText"
+        ]
+
+
+class MultiLingualAnnotator(Annotator):
+    """Custom Annotator class used to generate annotations."""
+
+    def __init__(self, app_path, config=None):
+        """Initializes an annotator.
+
+        Args:
+            app_path (str): The location of the MindMeld app
+            config (dict, optional): A config object to use. This will
+                override the config specified by the app's config.py file.
+        """
+        super().__init__(app_path=app_path, config=config)
+        self.language = self.config.get("language", "en")
+        self.translator = self.get_translator()
+        self.en_annotator = SpacyAnnotator(app_path)
+
+    def get_translator(self):
+        """Creates a Translator object based on the translator class specified in the config.
+        If a translator is not specified, Duckling heuristics will be used.
+
+        Returns:
+            translator (Translator): Translator specified in the config, None if not specified.
+        """
+        if self.config.get("translator"):
+            translator_class_name = self.config.get("translator", "GoogleTranslator")
```

Comment:
Why do we do this: `self.config.get("translator", "GoogleTranslator")`? This would return `GoogleTranslator` if `translator ` key is none.

Reply:
If translator is None, NoOpTranslator() is now selected as the default